### PR TITLE
[AMD][Kimi K2.5 Day 0] ROCm: route W4A16 MoE to Triton and fix packed-weight loading

### PR DESCRIPTION
**Pull request by jhinpan**
## Motivation
 As issue #17854 
On ROCm, `CompressedTensorsWNA16MoEMethod` currently routes to Marlin kernels by default. Marlin is NVIDIA‑only, which breaks Kimi‑K2.5 (native INT4) on MI300X. This patch dispatches ROCm to Triton and fixes the weight‑loading transpose path to avoid shape mismatches.

  ## Modifications
- `python/sglang/srt/layers/quantization/compressed_tensors/compressed_tensors_moe.py`
  - Add `CompressedTensorsWNA16TritonMoEMethod` and convert weights/scales to Triton layout.
  - Dispatch to Triton when `is_hip()` is true.
- `python/sglang/srt/layers/moe/fused_moe_triton/layer.py`
  - Include `CompressedTensorsWNA16TritonMoEMethod` in packed‑weight transpose checks.

## Accuracy Tests

tested on gsm8k

image

## Benchmarking and Profiling

Benchmark results here: notion.so/Kimi-K2-5-on-MI300X-2f5651cb22e580cb9395d6169ee59d66?pvs=73

## Checklist

- [x] Format your code according to the [Format code with pre-commit](https://docs.sglang.io/developer_guide/contribution_guide.html#format-code-with-pre-commit).
- [ ] Add unit tests according to the [Run and add unit tests](https://docs.sglang.io/developer_guide/contribution_guide.html#run-and-add-unit-tests).
- [ ] Update documentation according to [Write documentations](https://docs.sglang.io/developer_guide/contribution_guide.html#write-documentations).
- [ ] Provide accuracy and speed benchmark results according to [Test the accuracy](https://docs.sglang.io/developer_guide/contribution_guide.html#test-the-accuracy) and [Benchmark the speed](https://docs.sglang.io/developer_guide/contribution_guide.html#benchmark-the-speed).
- [ ] Follow the SGLang code style [guidance](https://docs.sglang.io/developer_guide/contribution_guide.html#code-style-guidance).

## Review Process

1. Ping Merge Oncalls to start the PR flow. See the [PR Merge Process](https://github.com/sgl-project/sglang/blob/main/.github/MAINTAINER.md#pull-request-merge-process).
2. Get approvals from [CODEOWNERS](https://github.com/sgl-project/sglang/blob/main/.github/CODEOWNERS) and other reviewers.
3. Trigger CI tests with [comments](https://docs.sglang.io/developer_guide/contribution_guide.html#how-to-trigger-ci-tests) or contact authorized users to do so.
   - `/tag-run-ci-label`, `/rerun-failed-ci`, `/tag-and-rerun-ci`
4. After green CI and required approvals, ask Merge Oncalls to merge.

### CODE REVIEW

**Review by HaiShaw — APPROVED** [review]
1. This should for MI350/355 as well
2. Can we add MoE triton tuning later?

File: python/sglang/srt/layers/quantization/compressed_tensors/compressed_tensors_moe.py [hunk 1]
```diff
@@ -1379,6 +1384,70 @@ def apply(
...
+    Inherits weight creation from CompressedTensorsWNA16MoEMethod but converts
+    weights to the uint8-packed format expected by the Triton fused MoE kernel
+    instead of the Marlin-specific format.
+    """
+
+    def process_weights_after_loading(self, layer: torch.nn.Module) -> None:
+        if getattr(layer, "is_triton_converted", False):
+            return
+
+        num_experts = layer.w13_weight_packed.shape[0]
```
Comment (kkHuang-amd) [inline python/sglang/srt/layers/quantization/compressed_tensors/compressed_tensors_moe.py:1399]: I don't see this variable "num_experts" is used in this function block, does it still need to exist?

Comment (jhinpan) [reply python/sglang/srt/layers/quantization/compressed_tensors/compressed_tensors_moe.py:1399]: Thanks for this nit! Will clean it today
